### ISSUE TITLE
[Bug]: Carbon/react ComboBox onChange prop is not getting called after production build. (tested in both React and Next JS).

### ISSUE THREAD

**Issue by sojinantony01** [carbon-design-system/carbon#18193]
### Package

@carbon/react

### Browser

Chrome

### Package version

1.70.0

### React version

18.3.1

### Description

Carbon/react ComboBox onChange prop is not getting called after production build. (tested in both React and Next JS).
Tested same in carbon react versions 1.69.0  1.70.0  1.71.1.  Found issue in all three versions.

### Reproduction/example

It is found only in production version.

### Steps to reproduce

- Create react application

```

import { ComboBox } from '@carbon/react';

function App() {
    const filterItems = (menu) => {
      return menu?.item?.toLowerCase().includes(menu?.inputValue?.toLowerCase());
    };
    const onChange = (e) => {
      console.log("onCHange called", e)
    }
  return (
    <div className="App">
      <ComboBox allowCustomValue shouldFilterItem={filterItems} onChange={onChange} id="carbon-combobox" items={['Apple', 'Orange', 'Banana', 'Pineapple', 'Raspberry', 'Lime']} titleText="ComboBox title" helperText="Combobox helper text" />
    </div>
  );
}

export default App;
```

- Render this component 

- Build the application and serve the build version
- Check on change calls 

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by AlanGreene** [comment]
Duplicate of https://github.com/carbon-design-system/carbon/issues/18145 maybe?

**Comment by sojinantony01** [comment]
Yes, it looks like duplicate of https://github.com/carbon-design-system/carbon/issues/18145
Closing as other issue already merged and fix available in [1.72.0-rc.0](https://www.npmjs.com/package/@carbon/react/v/1.72.0-rc.0)
